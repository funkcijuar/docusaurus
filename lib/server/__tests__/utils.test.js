/**
 * Copyright (c) 2017-present, Facebook, Inc.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
const path = require('path');
const fs = require('fs');
const utils = require('../utils');

describe('server utils', () => {
  test('minify css', () => {
    const testCss = fs.readFileSync(
      path.join(__dirname, '__fixtures__', 'test.css'),
      'utf8'
    );
    const notCss = fs.readFileSync(
      path.join(__dirname, '__fixtures__', 'test.md'),
      'utf8'
    );
    utils.minifyCss(testCss).then(css => expect(css).toMatchSnapshot());
    utils.minifyCss(notCss).catch(e => expect(e).toMatchSnapshot());
  });
});
