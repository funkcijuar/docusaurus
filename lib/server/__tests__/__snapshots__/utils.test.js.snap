// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`server utils minify css 1`] = `".hljs{margin-left:-15px;margin-right:-15px;border:1px solid #eee;border-radius:6px;padding:15px;font-size:15px;max-width:50rem}.hljs.javascript{background-color:rgba(247,223,30,.03)}.hljs .comment{opacity:.7}"`;

exports[`server utils minify css 2`] = `[Error: Unexpected "space" found.]`;
